viadot/sources/supermetrics.py

```diff
@@ -0,0 +1,164 @@
+import json
+import urllib
+from copy import deepcopy
+from typing import Any, Dict, List
+
+import numpy as np
+import pandas as pd
+
+from ..config import get_source_credentials
+from ..exceptions import CredentialError
+from ..utils import handle_api_response
+from .base import Source
+
+
+class Supermetrics(Source):
+    """
+    A class implementing the Supermetrics API.
+
+    Documentation for this API is located at: https://supermetrics.com/docs/product-api-getting-started/
+    Usage limits: https://supermetrics.com/docs/product-api-usage-limits/
+
+    Parameters
+    ----------
+    query_params : Dict[str, Any], optional
+        The parameters to pass to the GET query.
+        See https://supermetrics.com/docs/product-api-get-data/ for full specification,
+        by default None
+    """
+
+    API_ENDPOINT = "https://api.supermetrics.com/enterprise/v2/query/data/json"
+
+    def __init__(self, *args, query_params: Dict[str, Any] = None, **kwargs):
+        DEFAULT_CREDENTIALS = get_source_credentials("supermetrics")
+        credentials = kwargs.pop("credentials", DEFAULT_CREDENTIALS)
+
+        super().__init__(*args, credentials=credentials, **kwargs)
+
+        if not self.credentials:
+            self.logger.debug(
+                "Credentials not specified. Falling back to `boto3` default credentials."
+            )
+
+        self.query_params = query_params
+
+    @classmethod
+    def get_params_from_api_query(cls, url: str) -> Dict[str, Any]:
+        """Returns parmeters from API query in a dictionary"""
```

Comment:
I think we can standardize docstrings as in other sources https://github.com/dyvenia/viadot/blob/9239fcbf185f93230407fd16b4e96cd081e3d09b/viadot/sources/redshift_spectrum.py#L109